Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -115,28 +146,42 @@ class BrowserViewController: UIViewController {
         }
 
         // Setup the bottom toolbar
-        toolbar.snp_remakeConstraints { make in
-            make.edges.equalTo(self.footerBackground)
+        toolbar?.snp_remakeConstraints { make in
+            make.edges.equalTo(self.footerBackground!)
             make.height.equalTo(AppConstants.ToolbarHeight)
-            return
         }
 
         adjustFooterSize()
-        footerBackground.snp_remakeConstraints { make in
+        footerBackground?.snp_remakeConstraints { make in
             make.bottom.left.right.equalTo(self.footer)
             make.height.equalTo(AppConstants.ToolbarHeight)
         }
         urlBar.setNeedsUpdateConstraints()
+
+        // Remake constraints even if we're already showing the home controller.
+        // The home controller may change sizes if we tap the URL bar while on about:home.
+        homePanelController?.view.snp_remakeConstraints { make in
+            make.top.equalTo(self.urlBar.snp_bottom)
+            make.left.right.equalTo(self.view)
+            let url = self.tabManager.selectedTab?.url
+            if url?.absoluteString == HomeURL {
```

Comment:
Unfortunately, the condition isn't this straightforward. As the comment above says, "The home controller may change sizes if we tap the URL bar while on about:home.". That is, if we're on about:home, we show the bottom toolbar since we treat it like any other web page. But once we tap the URL bar, we're in "home mode", so the toolbar should disappear.

That was the reason the `#inline: Bool` argument was added to `showHomePanelController`. The location alone isn't enough information for us to make that decision.